src/entities/modal-window/lib/modal.ts

```diff
@@ -0,0 +1,61 @@
+import { computed, ref, shallowRef } from 'vue'
+import type { Component, ComputedRef } from 'vue'
+
+import type { Modal } from '../types'
+
+const innerComponent = shallowRef()
+const title = ref('')
+const html = document.querySelector('.page')
+
+if (!html) {
+  throw new Error(`HTML element is undefined`)
+}
+
+export const open = ref(false)
+
+export const modal: ComputedRef<Modal> = computed(() => ({
```

Comment:
Так же самая запись, но покороче.

```suggestion
export const modal = computed<Modal>(() => ({
```